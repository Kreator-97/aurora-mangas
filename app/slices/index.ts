export { default as uiSlice } from './uiSlice'
export { default as authSlice } from './authSlice'
export { default as shoppingCartSlice } from './shoppingCartSlice'
