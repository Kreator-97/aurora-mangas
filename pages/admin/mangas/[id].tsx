import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'
import Image from 'next/image'

import { AppLayout } from '../../../layouts'
import { dbItems, dbMangas, dbUsers } from '../../../database'
import { CustomRow } from '../../../components'
import { Manga } from '../../../interfaces'
import { formatPrice } from '../../../util'

interface Props {
  manga: Manga & { soldUnits: number }
}

const MangaDetailsPage:NextPage<Props> = ({manga}) => {
  return (
    <AppLayout title="Detalles">
      <div className='py-2'>
        <h1 className='title mb-4'>Detalles del producto</h1>
        <div
          className='mx-auto gap-4'
          style={{
            display: 'grid',
            gridTemplateColumns: '200px 1fr',
            maxWidth: '600px',
          }}>
          <div>
            <Image src={manga.imgURL} layout="fixed" width={200} height={300} />
          </div>
          <div>
            <CustomRow name={'Titulo'} value={manga.title} />
            <CustomRow name={'Serie'} value={ manga.serie.name } />
            <CustomRow name={'NÃºmero'} value={ manga.number } />
            <CustomRow name={'Publicado'} value={ manga.published } />
            <CustomRow name={'Precio'} value={ formatPrice(manga.price) } />
            <CustomRow name={'Stock'} value={ manga.stock } />
            <CustomRow name={'Unidades vendidas'} value={ manga.soldUnits } />
          </div>
        </div>
      </div>
    </AppLayout>
  )
}

export default MangaDetailsPage

export const getServerSideProps: GetServerSideProps = async (ctx) => {

  const session = await getSession(ctx)

  if( !session ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const isValid = await dbUsers.validateRole(session.user.id, ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const id = ctx.query.id?.toString()

  const manga = await dbMangas.getMangaById(id || '')
  const soldUnits = await dbItems.getSoldUnitsByMangaId(id || '')

  return {
    props: {
      manga: {
        ...manga,
        soldUnits
      }
    }
  }
}
