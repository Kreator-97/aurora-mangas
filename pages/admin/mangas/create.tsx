import { FormEvent, useState } from 'react'
import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'

import { AppLayout } from '../../../layouts'
import { dbSeries, dbUsers } from '../../../database'
import { useForm } from '../../../hooks'
import { Serie } from '../../../interfaces'

const CLOUDINARY_DOMAIN = 'https://res.cloudinary.com'

const formValidations = {
  serie: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  imgURL: {
    validation: (value:string) => value.trim() !== '' && value.startsWith(CLOUDINARY_DOMAIN) ,
    message: 'Este campo debe de ser una URL de Cloudinary válida',
  },
  number: {
    validation: (value:string) => ( !isNaN(Number(value)) ) && value !== '',
    message: 'Este campo debe de ser un número válido',
  },
  price: {
    validation: (value:string) => ( !isNaN(Number(value)) ) && value !== '',
    message: 'Este campo debe de ser un número válido',
  },
}

interface Props {
  series: Serie[];
}

const AdminCreateSerie:NextPage<Props> = ({series}) => {
  const [showErrors, setShowErrors] = useState(false)
  const { imgURL, number, price, title, serie, formState, errors, onInputChange } = useForm({
    serie: '',
    imgURL: '',
    number : '',
    price : '',
    title : '',
  }, formValidations)


  const onSave = (e:FormEvent) => {
    e.preventDefault()
    if( Object.values(errors).some( value => value !== null) ) {
      setShowErrors(true)
      return
    }
    console.log('formulario válido')
    console.log('Hay que realizar la inserción')
    console.log({formState})
  }

  return (
    <AppLayout title="Crear manga" maxWidth='md'>
      <h1 className='title'>Agregar nuevo manga</h1>
      <form className='mx-auto mb-4 max-w-lg' onSubmit={onSave}>
        <div className='flex flex-col gap-4 mb-4'>

          <div className='flex flex-col gap-2'>
            <label htmlFor="serie" className='text-lg'>
              Seleccionar serie
            </label>
            <select
              name="serie"
              id="serie"
              className='input bg-white'
              value={ serie }
              onChange={ onInputChange }
            >
              <option
                value=""
                defaultChecked
                disabled>Selecciona un autor
              </option>
              {
                series.map((serie) => {
                  return (
                    <option key={serie.id} value={serie.id}>{serie.name}</option>
                  )
                })
              }
            </select>
            { (errors.serie && showErrors) && (<p className='text-error'>{errors.serie}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="number" className='text-lg'>
              Escribe el número de volumen
            </label>
            <input
              min={0}
              type="number"
              name="number"
              id="number"
              className='input'
              placeholder='Número'
              value={ number }
              onChange={ onInputChange }
            />
            { (errors.number && showErrors) && (<p className='text-error'>{errors.number}</p>) }
          </div>
          
          <div className='flex flex-col gap-2'>
            <label htmlFor="price" className='text-lg'>
              Escribe el precio unitario
            </label>
            <input
              min={0}
              type="number"
              name="price"
              id="price"
              className='input'
              placeholder='Escribe el precio unitario'
              value={ price }
              onChange={ onInputChange }
            />
            { (errors.price && showErrors) && (<p className='text-error'>{errors.price}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="imgURL" className='text-lg'>
              Coloca la URL de la serie (cloudinary)
            </label>
            <input
              type="text"
              name="imgURL"
              id="imgURL"
              className='input'
              placeholder='Coloca una url de cloudinary'
              value={ imgURL }
              onChange={ onInputChange }
            />
            { (errors.imgURL && showErrors) && (<p className='text-error'>{errors.imgURL}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="title" className='text-lg'>
              Escribe un titulo (opcional)
            </label>
            <input
              type="text"
              name="title"
              id="title"
              className='input'
              placeholder='Titulo opcional'
              value={ title }
              onChange={ onInputChange }
            />
          </div> 
        </div>
        <button className='btn bg-accent w-full'>Guardar</button>
      </form>
    </AppLayout>
  )
}

export default AdminCreateSerie

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)
  
  const isValid = dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const series = await dbSeries.getSeries()

  return {
    props: {
      series,
    }
  }
}
