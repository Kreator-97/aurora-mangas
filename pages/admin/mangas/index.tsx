import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'
import Link from 'next/link'

import { dbMangas, dbUsers } from '../../../database'
import { Manga } from '../../../interfaces'
import { AppLayout } from '../../../layouts'
import { Table } from '../../../components'

interface Props {
  mangas: Manga[]
}

const AdminProductPage: NextPage<Props> = ({mangas}) => {
  return (
    <AppLayout title="Series | Admin" maxWidth='lg'>
      <div className='px-2'>
        <div className='flex justify-between items-center py-4'>
          <div className='w-36'></div>
          <h1 className='title'>Mangas</h1>
          <Link href='/admin/mangas/create'>
            <button className='btn bg-accent text-sm'>Agregar manga</button>
          </Link>
        </div>
        {
          mangas.length === 0 
            ? (
              <h2 className='text-lg text-center'>No existen datos de Mangas</h2>
            )
            : (
              <Table
                columns={['Id', 'Nombre', '# Volumen', 'Precio Unitario($)', 'PublicaciÃ³n']}
              >
                {
                  mangas.map((manga) => {
                    return (
                      <tr key={manga.id}>
                        <td>{manga.id}</td>
                        <td>{manga.serie.name}</td>
                        <td>{manga.number}</td>
                        <td>{manga.price}</td>
                        <td>{manga.published}</td>
                      </tr>
                    )
                  })
                }
              </Table>
            )
        }
      </div>
    </AppLayout>
  )
}

export default AdminProductPage

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)

  const isValid = await dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const mangas = await dbMangas.getAllMangas()

  return {
    props: {
      mangas,
    }
  }
}
