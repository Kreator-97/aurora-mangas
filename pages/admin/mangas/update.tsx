import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'
import toast from 'react-hot-toast'
import { useMutation } from '@apollo/client'

import { AppLayout } from '../../../layouts'
import { UPDATE_MANGA } from '../../../graphql/client'
import { dbMangas, dbSeries, dbUsers } from '../../../database'
import { Manga, Serie } from '../../../interfaces'
import { FormCreateManga } from '../../../components'

interface Props {
  series: Serie[];
  manga: Manga;
}

const AdminUpdateManga:NextPage<Props> = ({series, manga}) => {
  const [ updateManga ] = useMutation(UPDATE_MANGA)

  const onSave = async (formState:FormCreateManga) => {
    const { imgURL, month, number, price, serie, title, year } = formState
    const monthPrefixed = Number(month) < 10 ? `0${Number(month)}` : month
    const published = `${year}/${monthPrefixed}/01`

    try {
      const { data } = await updateManga({
        variables: {
          mangaId: manga.id,
          manga: {
            serieId: serie, title, number, imgURL, price: Number(price), published
          }
        }
      })
      
      const message = data.updateManga.message

      if( !data.updateManga.ok ) {
        throw new Error(message)
      }
      
      toast.success(message)
    } catch (error) {
      toast.error((error as {message: string}).message)
      console.log(error)
    }
  }

  return (
    <AppLayout title="Actualizar manga" maxWidth='md'>
      <div className='px-2'>
        <h1 className='title'>Actualizar manga</h1>
        <FormCreateManga series={series} onSubmit={onSave} manga={manga} resetOnSubmit={false}/>
      </div>
    </AppLayout>
  )
}

export default AdminUpdateManga

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)

  const isValid = dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const id = ctx.query.id?.toString()

  const [series, manga] = await Promise.all([
    await dbSeries.getSeries(),
    await dbMangas.getMangaById(id || ''),
  ])

  return {
    props: {
      series,
      manga
    }
  }
}
