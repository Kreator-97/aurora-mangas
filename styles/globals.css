@tailwind base;
@tailwind components;
@tailwind utilities;
@tailwind variants;

:root {
  box-sizing: border-box;
  --bg-color: #001E31;
  --bg-color2: #00474F;
  --stroke-color: #07695b70;
  --success: #21C728;
  --success-dark: #118A16;
  --dark: #333;
  --light: #eee;
  --header-height: 88px;
  --sm: 480px;
  --md: 720px;
  --lg: 960px;
  --xl: 1200px;
  --max: 1440px;
}

html {
  color: var(--light);
  background-image: linear-gradient( var(--bg-color), var(--bg-color2));
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
}

.main {
  min-height: calc(100vh -var(--header-height));
  margin: var(--header-height) auto 0;
}

.title {
  @apply text-xl md:text-2xl lg:text-3xl text-center py-2;
}

.btn {
  @apply px-2 py-1 font-normal text-light rounded transition-colors border border-solid border-light;
}

.btn.ghost {
  @apply text-accent bg-opacity-0 border-accent;
}

.bg-success-gradient {
  background-image: linear-gradient( var(--success), var(--success-dark))!important;
}

.btn.ghost:hover {
  @apply bg-light;
}

.active {
  @apply bg-success;
}

@media screen and (min-width: 720px) {
  .main {
    padding: 0 .5rem;
  }
}
