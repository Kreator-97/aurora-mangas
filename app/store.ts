import { configureStore } from '@reduxjs/toolkit'
import { uiSlice, authSlice, shoppingCartSlice } from './slices'

export const store = configureStore({
  reducer: {
    ui: uiSlice,
    auth: authSlice,
    cart: shoppingCartSlice
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
