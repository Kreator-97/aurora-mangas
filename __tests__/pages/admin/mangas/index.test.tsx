import { expect } from '@jest/globals'
import { render, screen } from '@testing-library/react'
import { useSession } from 'next-auth/react'
import { Provider } from 'react-redux'
import { store } from '../../../../app/store'
import { mangas } from '../../../fixtures/db'

import { default as AdminMangaPage } from '../../../../pages/admin/mangas/index'

jest.mock('next/router', () => ({
  useRouter: () => ({query: { q: '' }})
}))

jest.mock('next-auth/react')

describe('tests on AdminManga page', () => {

  (useSession as jest.Mock).mockReturnValue({
    useSession: () => ({status: 'unauthenticated', data: null})
  })

  test('should to match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <AdminMangaPage mangas={mangas} />
      </Provider>
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Mangas'))
    expect(screen.getByText('Agregar manga'))
  })

  test('should to show a message when there are not mangas', () => {
    render(
      <Provider store={store}>
        <AdminMangaPage mangas={[]} />
      </Provider>
    )

    expect(screen.getByText('No existen datos de Mangas'))
  })
})
