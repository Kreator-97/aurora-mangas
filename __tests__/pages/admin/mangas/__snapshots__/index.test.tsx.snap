// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on AdminManga page should to match with snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="content"
    >
      <svg
        color="white"
        cursor="pointer"
        data-testid="menu"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0.5"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 150px; height: 30px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Aurora Mangas Logo"
          class="cursor-pointer"
          data-nimg="fixed"
          data-testid="app-logo"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="relative cursor-pointer"
        data-testid="cart-menu"
      >
        <svg
          color="white"
          cursor="pointer"
          data-testid="shopping-cart-menu"
          fill="currentColor"
          height="32"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="w-6 h-6 text-center font-semibold bg-accentDark bg-opacity-95 rounded-full absolute -top-2 -right-2"
        >
          0
        </span>
      </div>
      <form
        class="flex col-span-2"
        data-testid="search-form"
      >
        <input
          class="w-full rounded border-cyan-400 border p-1 text-sm text-dark"
          data-testid="search-input"
          placeholder="Encuentra tu manga favorito. Ejemplo: one punch man #1"
          type="search"
          value=""
        />
      </form>
      <svg
        color="white"
        cursor="pointer"
        data-testid="search-icon"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16l2.879-2.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242zM21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </nav>
  <div
    class="min-h-screen max-w-[360px] bg-dark fixed top-0 left-0 w-full z-50 p-4 transition-transform -translate-x-full z-[1000]"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <div
        class="w-8"
      />
      <h2
        class="text-2xl"
      >
        Bienvenido
      </h2>
      <svg
        class="cursor-pointer"
        data-testid="close-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="text-center mb-2"
    >
      Ingresa para gestionar tu cuenta
    </p>
    <div
      class="flex gap-2 mb-4"
    >
      <button
        class="btn bg-accent w-full"
      >
        Ingresar
      </button>
      <button
        class="btn ghost w-full"
      >
        Crear cuenta
      </button>
    </div>
  </div>
  <div
    class="fixed top-0 right-0 min-h-screen h-screen min-w-[360px] bg-dark z-50 transition-transform translate-x-full"
  >
    <div
      class="flex justify-between items-center border-b border-strokeLight p-4"
      style="height: 65px;"
    >
      <svg
        cursor="pointer"
        data-testid="close-shopping-cart-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <h2
        class="text-center text-2xl"
      >
        Carrito de compra
      </h2>
      <div
        class="w-7"
      />
    </div>
    <div
      class="flex flex-col gap-4 items-center overflow-y-scroll cartlist-items"
      style="padding-top: 1rem; padding-bottom: 1rem;"
    >
      <h2
        class="text-center"
      >
        No hay elementos en el carrito de compra
      </h2>
    </div>
  </div>
  <main
    class="main"
    style="max-width: var(--lg);"
  >
    <div
      class="px-2"
    >
      <div
        class="flex justify-between items-center py-4"
      >
        <div
          class="w-36"
        />
        <h1
          class="title"
        >
          Mangas
        </h1>
        <button
          class="btn bg-accent text-sm"
        >
          Agregar manga
        </button>
      </div>
      <table
        class="table"
      >
        <thead>
          <tr>
            <th
              data-testid="column"
            >
              Id
            </th>
            <th
              data-testid="column"
            >
              Nombre
            </th>
            <th
              data-testid="column"
            >
              # Volumen
            </th>
            <th
              data-testid="column"
            >
              Precio Unitario($)
            </th>
            <th
              data-testid="column"
            >
              Stock
            </th>
            <th
              data-testid="column"
            >
              Publicaci√≥n
            </th>
            <th
              data-testid="column"
            >
              Editar
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              02e7e70f-0d0d-43c2-86e6-34e75b7025e3
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              12
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/09/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=02e7e70f-0d0d-43c2-86e6-34e75b7025e3"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              19fbddbe-2157-45e5-a58b-da2670a25bf8
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              8
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/05/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=19fbddbe-2157-45e5-a58b-da2670a25bf8"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              20bcecfd-5599-4027-921a-ab4e544875a8
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              5
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/02/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=20bcecfd-5599-4027-921a-ab4e544875a8"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              21299faf-231f-48e1-9e54-975ce7d68716
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              2
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2021/11/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=21299faf-231f-48e1-9e54-975ce7d68716"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              2711e383-e7d3-4169-bd29-e7bb98f7375c
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              1
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2021/10/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=2711e383-e7d3-4169-bd29-e7bb98f7375c"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              3c5fdf37-ea75-4cad-a701-d7ec98edbaa2
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              10
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/07/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=3c5fdf37-ea75-4cad-a701-d7ec98edbaa2"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              51755e4c-9523-4c68-80fb-cee7dd080fe8
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              7
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/04/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=51755e4c-9523-4c68-80fb-cee7dd080fe8"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              8996f64d-5e5d-45bd-8041-acc331e47e10
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              4
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/01/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=8996f64d-5e5d-45bd-8041-acc331e47e10"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              c3d1c41a-38a0-4245-bdb4-8b680fe4d917
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              11
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/08/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=c3d1c41a-38a0-4245-bdb4-8b680fe4d917"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              c6d9a827-9f89-431a-b29f-ce9714cf8a40
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              9
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/06/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=c6d9a827-9f89-431a-b29f-ce9714cf8a40"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              d03805c0-2f4a-468b-91aa-4997d6d44923
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              3
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2021/12/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=d03805c0-2f4a-468b-91aa-4997d6d44923"
              >
                Editar üìù
              </a>
            </td>
          </tr>
          <tr>
            <td>
              dd6e8fa8-a111-4758-8b52-b1e2329dace6
            </td>
            <td>
              One Punch Man
            </td>
            <td>
              6
            </td>
            <td>
              $99.00
            </td>
            <td>
              10
            </td>
            <td>
              2022/03/01
            </td>
            <td>
              <a
                class="btn bg-accent"
                href="/admin/mangas/update?id=dd6e8fa8-a111-4758-8b52-b1e2329dace6"
              >
                Editar üìù
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </main>
  <footer
    class="min-h-[150px] border-t border-solid border-stroke"
  >
    <h2
      class="text-xl md:text-3xl text-center p-4 mt-8"
    >
      S√≠guenos en nuestras redes sociales
    </h2>
    <div
      class="flex justify-center gap-8"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="facebook"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 36px; height: 36px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="twitter"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </footer>
</div>
`;
